cloudant-client/src/main/java/com/cloudant/client/api/query/JsonIndex.java

```diff
@@ -0,0 +1,205 @@
+                    field = new Field(fieldEntry.getKey(), context.<Sort.Order>deserialize
+                            (fieldEntry.getValue(), SORT_ORDER));
+                }
+                return field;
+            }
+
+            @Override
+            public JsonElement serialize(Field src, Type typeOfSrc, JsonSerializationContext
+                    context) {
+                if (src.getOrder() != null) {
+                    JsonObject field = new JsonObject();
+                    field.add(src.getName(), context.serialize(src.getOrder()));
+                    return field;
+                } else {
+                    return new JsonPrimitive(src.getName());
+                }
+            }
+        }
+    }
+
+    /**
+     * Class for building a definition for a JSON type index.
+     */
+    public static class Builder extends com.cloudant.client.internal.query.Builder<JsonIndex,
+            Definition, Builder, Field> {
+
+        private Builder() {
+            // Prevent instantiation except by static JsonIndex.builder()
+        }
+
+        protected JsonIndex newInstance() {
+            return new JsonIndex();
+        }
+
+        @Override
+        protected Builder returnThis() {
+            return this;
+        }
+
+        /**
+         * Configure the name of the index, if not set a name will be generated.
+         *
+         * @param indexName name of the index
+         * @return the builder for chaining
+         */
+        public Builder name(String indexName) {
+            return super.name(indexName);
+        }
+
+        /**
+         * Configure the design document name, if not set a new design document will be created with
+         * a generated name.
+         *
+         * @param designDocumentId design document ID (the _design prefix is added if not supplied)
+         * @return the builder for chaining
+         */
+        public Builder designDocument(String designDocumentId) {
+            return super.designDocument(designDocumentId);
+        }
+
+        /**
+         * <p>
+         * Configure a selector string to choose documents that should be added to the index.
+         * </p>
+         * <p>
+         * The easiest way of obtaining a selector string is to obtain a {@link Selector} from an
+         * {@link Operation} or {@link Expression} and call {@code toString} on the resulting
+         * {@link Selector}.
+         * </p>
+         * @param selector string representation of a JSON object describing criteria used to add
+         *                 documents to index
+         * @return the builder for chaining
+         * @see Selector
+         * @see <a
+         * href="https://console.bluemix.net/docs/services/Cloudant/api/cloudant_query.html#selector-syntax"
+         * target="_blank">selector syntax</a>
+         */
+        @Override
+        public Builder partialFilterSelector(String selector) {
```

Comment:
I'm leaning back towards using the `Selector` type here, given that `Database#createIndex(..)` accepts a `String` for the body there is still a way to workaround any selector limitations without having to accept the `String` here.

Reply:
Done in 63531d3